processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -263,4 +264,24 @@ public static Float convertObjectToFloat(Object valObj)
       throw new ParseException("Unknown type[%s]", valObj.getClass());
     }
   }
+
+  public static Long getIntegralFromFloatString(String floatStr)
+  {
+    BigDecimal convertedBD;
+    try {
+      convertedBD = new BigDecimal(floatStr);
+    } catch (NumberFormatException nfe) {
+      return null;
+    }
+
+    long asLong;
+    try {
+      asLong = convertedBD.longValueExact();
+    } catch (ArithmeticException ae) {
+      // indicates there was a non-integral part, or the BigDecimal was too big for a long
```

Comment:
I'm not sure it's ok for just ignoring this exception. Maybe is it better to indicate the parse failure to users?